### DISCUSSION TITLE
Token-exchange  SDK API or REST API in v21

### DISCUSSION THREAD

**Discussion by Marimuthu14** [keycloak/keycloak#20074] [Q&A]
Pre-requisites : 
Fine grained auth / token exchange  features are enabled. 
Use case : 
Exchange token between two pubic clients of same realm. 
Expected behavior : 
This is been working in v14, now we are upgrading to v20 or v21, throws unauthorized error on v20 and v21. 
We figured out token exchange works when we create a token-exchange permission through admin client. Hence I am trying to make it work programmatically which is were I am struck.
I am looking for keycloak SDK API to create token-exchange permission with scope based policy to perform the token-exchange.
 programmatically in latest versions. (v20 or v21) from admin client. I couldn’t find a API to persist the created ClientPolicyRepresentation / ScopePermissionRepresentation entities from the admin client.  

Any advise would be appreciated.

Steps performed: 
1. create public client ( Source Client) 
2. create another public client  in the same realm(Target Client ) . 
3. now enable the fine grained permission toggle button in permissions tab 
4. Now select token-exchange from the permission list and create a permission to perform token exchange
5. Click on client details and go to authorization menu
6. create a client policy type for the clients : Source client 
7. Now associate the policy created in above step with permission created in step 4. 
8. save the permission with policy. 
9. now perform token-exchange with generated public access token for source client. 
10. Throws unauthorized error. 
11. Expected behavior is the valid token in response. 

I am trying this programmatically through SDK and through rest API which throws 403 response error

**Comment by ffroliva** [comment]
Any update on this? Where you able to programmatically configure client-to-client (internal) token exchange?

I am working with keycloak 25.0.1 but I  am facing difficulties.
